megamek/src/megamek/client/ui/SharedUtility.java

```diff
@@ -905,19 +905,12 @@ public static String[] getDisplayArray(List<? extends Targetable> entities) {
         return retVal;
     }
 
-    public static @Nullable Targetable getTargetPicked(List<? extends Targetable> targets,
-                                                       String input) {
-        if (input == null) {
+    public static @Nullable Targetable getTargetPicked(@Nullable List<? extends Targetable> targets,
+                                                       @Nullable String chosenDisplayName) {
+        if ((chosenDisplayName == null) || (targets == null)) {
             return null;
+        } else {
+            return targets.stream().filter(t -> chosenDisplayName.equals(t.getDisplayName())).findAny().orElseGet(null);
```

Comment:
```suggestion
            return targets.stream().filter(t -> chosenDisplayName.equals(t.getDisplayName())).findAny().orElse(null);
```
This will NPE if called with null. Optional::orElse is the proper method to call.